### ISSUE TITLE
Found > 1 "module.xml" for the module

### ISSUE THREAD

**Issue by friendscottn** [DrewML/baler#11]
These Mirasvit modules are really throwing some curve balls.
```
Error: Found > 1 "module.xml" for the module at path "vendor/mirasvit/module-search"
    at findRealModuleRoot (/Users/scottfriend/Sites/baler/src/magentoFS.ts:247:15)
    at processTicksAndRejections (internal/process/task_queues.js:89:5)
    at getModuleConfig (/Users/scottfriend/Sites/baler/src/magentoFS.ts:204:28)
    at async Promise.all (index 151)
    at getComposerComponents (/Users/scottfriend/Sites/baler/src/magentoFS.ts:332:18)
    at async Promise.all (index 0)
    at Object.getComponents (/Users/scottfriend/Sites/baler/src/magentoFS.ts:62:37)
    at async Promise.all (index 1)
    at Object.collectStoreData (/Users/scottfriend/Sites/baler/src/collectStoreData.ts:15:58)
    at Object.run (/Users/scottfriend/Sites/baler/src/cli.ts:20:19)
```

Here's the abbreviated directory structure of this module:
vendor/mirasvit/module-search
- CHANGELOG.md
- composer.json
- src
    - Search
        - ...
        - composer.json
        - etc
            - ...
            - module.xml
        - ...
        - registration.php
    - SearchElasticNative
        ...
        - composer.json
        - etc
            - ...
            - module.xml
        - ...
        - registration.php
    
I'm really not sure why it makes sense to have two Mage modules inside one composer module =/

**Comment by DrewML** [comment]
> These Mirasvit modules are really throwing some curve balls.

The modules are definitely doing a good job of proving all my assumptions wrong 😂. Not too surprised, though - Magento core doesn't impose a huge number of constraints on packages, and `baler` is still super new (appreciate you testing things out, btw).

Can you show me what the root `composer.json` looks like here? Curious if each module dir's `composer.json` has its own autoload for `registration.php`, or if there are multiple in the root.

**Comment by friendscottn** [comment]
The vendor/mirasvit/module-search/composer.json looks like:
```json
{
  "name": "mirasvit/module-search",
  "description": "Search",
  "require": {
    "magento/framework": "100.1.*|100.2.*|101.0.*|102.0.*",
    "mirasvit/module-core": "dev-master|>=1.2.75",
    "mirasvit/module-search-mysql": "dev-master|1.0.*"
  },
  "type": "magento2-module",
  "version": "1.0.122",
  "license": [
    "proprietary"
  ],
  "autoload": {
    "files": [
        "src/Search/registration.php",
        "src/SearchElasticNative/registration.php"
    ],
    "psr-4": {
      "Mirasvit\\Search\\": "src/Search",
      "Mirasvit\\SearchElasticNative\\": "src/SearchElasticNative"
    }
  }
}
```
I diffed the other two composer.json files and they are identical to the root (above):
- vendor/mirasvit/module-search/src/Search/composer.json
- vendor/mirasvit/module-search/src/SearchElasticNative/composer.json

So... that's not great =)

**Comment by Vinai** [comment]
In general it would probably be good to allow modules that follow the [Maven SDL](https://maven.apache.org/guides/introduction/introduction-to-the-standard-directory-layout.html)... I always thought Magento enforcing their own thing was bad. If I recall correctly, extensions submitted to the Marketplace have to follow the Magento recommendations.

I'm afraid you'll have to probably parse the `registration.php` files or the `composer.json` autoload section to build any directory paths... You probably are already doing that anyway for some things, or not?

**Comment by DrewML** [comment]
> I always thought Magento enforcing their own thing was bad

Yeah, I'm not a fan, but we're here now 😂

> I'm afraid you'll have to probably parse the registration.php files or the composer.json autoload section to build any directory paths... You probably are already doing that anyway for some things, or not?

Yeah, this seems like the most reliable way forward. 

**We already collect the autoload `registration.php entries**:

https://github.com/DrewML/baler/blob/104d5d8f8204a2fb67f619e7cd27e4eb0a97aa5e/src/magentoFS.ts#L392-L396

**and we have already started doing analysis of values in `registration.php`**:

https://github.com/DrewML/baler/blob/104d5d8f8204a2fb67f619e7cd27e4eb0a97aa5e/src/parseThemeIDFromRegistration.ts#L3-L11

**Comment by DrewML** [comment]
Instead of the fix mentioned above, #12 _should_ take care of this problem when it's finished.

**Comment by DrewML** [comment]
The work from #12 is now in `master`, and I'm getting the same results for a stock store that I did before, so things are looking good.

@friendscottn want to give it another spin?

**Comment by friendscottn** [comment]
Yep I think that did it. Nice work! That fixes #10 and this one
